src/realm/query_engine.cpp

```diff
@@ -17,9 +17,65 @@
  **************************************************************************/
 
 #include <realm/query_engine.hpp>
+#include <realm/column_table.hpp>
 
 using namespace realm;
 
+namespace realm {
+
+Columns<SubTable>::Columns(const Columns& other, QueryNodeHandoverPatches* patches)
+    : Subexpr2<SubTable>(other)
+    , m_link_map(other.m_link_map, patches)
+    , m_column_ndx(other.m_column_ndx)
+    , m_column(other.m_column)
+{
+    if (m_column && patches)
+        m_column_ndx = m_column->get_column_index();
+}
+
+void Columns<SubTable>::evaluate(size_t index, ValueBase& destination)
```

Comment:
Wait until you see my subtable query support. Then we can discuss if there is a more straight forward way to do things.